src/rpc-handler.ts

```diff
@@ -0,0 +1,212 @@
+import { JsonRpcProvider } from "@ethersproject/providers";
+import { LOCAL_HOST, networkRpcs, tokens, networkIds, networkNames, networkCurrencies, networkExplorers, Token } from "./constants";
+import { HandlerInterface, HandlerConstructorConfig } from "./handler";
+
+import { RPCService } from "./services/rpc-service";
+import { StorageService } from "./services/storage-service";
+
+export class RPCHandler implements HandlerInterface {
+  private static _instance: RPCHandler | null = null;
+  private _provider: JsonRpcProvider | null = null;
+  private _networkId: number;
+  private _env: string = "node";
+
+  private _cacheRefreshCycles: number = 10;
+  private _refreshLatencies: number = 0;
+  private _autoStorage: boolean = false;
+
+  private _runtimeRpcs: string[] = [];
+  private _latencies: Record<string | number, number> = {};
+
+  private _networkIds = networkIds;
+  private _tokens = tokens;
+  private _networkNames = networkNames;
+  private _networkCurrencies = networkCurrencies;
+  private _networkRpcs = networkRpcs;
+  private _networkExplorers = networkExplorers;
+
+  constructor(networkId: number, config?: HandlerConstructorConfig) {
+    this._networkId = networkId;
+    this._initialize(config);
+  }
+
+  public async getFastestRpcProvider(networkId: number): Promise<JsonRpcProvider> {
+    if (networkId === 31337) {
+      this._provider = new JsonRpcProvider(LOCAL_HOST, networkId);
+    } else if (networkId !== 31337 && this._provider?.connection.url.includes("localhost")) {
+      this._provider = await this.testRpcPerformance(networkId);
+    } else {
+      this._provider = await this.testRpcPerformance(networkId);
```

Comment:
else if and else are identical

Reply:
it's idiomatic to put the last else, as is, tho